torch/distributed/_tensor/_utils.py

```diff
@@ -77,6 +79,73 @@ def compute_local_offset(
         return tuple(local_offsets)
 
 
+def compute_local_shape_and_global_offset(
```

Comment:
I think the original two APIs are correlated: 1. compute_local_tensor_shape 2. compute_local_tensor_offset, the second is not related to any "local_offset" per-say, it should actually compute the global offset of its "local shard", maybe the names are misleading. Shall we consolidate this API into the above two in this PR?